backtrace_symbols function is not present on Musl based systems

Hello

Compiling p4c on e.g. Alpine which uses Musl instead of Glibc fails:
```
[  2%] Built target linkp4cebpf
Scanning dependencies of target p4c_driver
[  3%] Copying p4c driver
[  3%] Built target p4c_driver
In file included from /p4c/build/lib/unified_libp4ctoolkit_srcs_1.cpp:2:
/p4c/lib/backtrace.cpp: In function 'void backtrace_fill_stacktrace(std::string&, void* const*, int)':
/p4c/lib/backtrace.cpp:26:22: error: 'backtrace_symbols' was not declared in this scope
   26 |     char **strings = backtrace_symbols(backtrace, size);
      |                      ^~~~~~~~~~~~~~~~~
make[2]: *** [lib/CMakeFiles/p4ctoolkit.dir/build.make:82: lib/CMakeFiles/p4ctoolkit.dir/unified_libp4ctoolkit_srcs_1.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1191: lib/CMakeFiles/p4ctoolkit.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  5%] Linking CXX static library libgtest.a
[  5%] Built target gtest
make: *** [Makefile:160: all] Error 2
```

See e.g. https://www.openwall.com/lists/musl/2015/04/09/3

--- a/lib/backtrace.cpp
+++ b/lib/backtrace.cpp
@@ -43,12 +43,12 @@ void backtrace_fill_stacktrace(std::string &msg, void *const*backtrace, int size
 #endif
 }
 
-namespace std {
-
 #ifdef __GLIBC__
 /* DANGER -- overrides for glibc++ exception throwers to include a stack trace.
  * correct functions depends on library internals, so may not work on some versions
  * and will fail with non-GNU libc++. These function are declared in bits/functexcept.h. */
+
+namespace std {
 
 void __throw_bad_alloc() {
     throw backtrace_exception<bad_alloc>();

